File: game-template/tien-len-deno-ts/test/performance.java
public class performance {
    private static final long TOTAL_ITERATIONS = 1_000_000_000; // Số lần lặp (1 tỷ)

    public static long performHeavyTask() {
        long result = 0;
        for (long i = 0; i < TOTAL_ITERATIONS; i++) {
            result += i;
        }
        return result;
    }

    public static void main(String[] args) {
        long startTime = System.currentTimeMillis();
        long result = performHeavyTask();
        long endTime = System.currentTimeMillis();

        System.out.println("Kết quả: " + result);
        System.out.println("Thoi gian thuc thi: " + (endTime - startTime) + "ms");
    }
}

File: game-template/tien-len-deno-ts/test/performance.cpp
#include <iostream>
#include <chrono>

const long long totalIterations = 1000000000; // Số lần lặp (1 tỷ)

long long performHeavyTask() {
    long long result = 0;
    for (long long i = 0; i < totalIterations; i++) {
        result += i;
    }
    return result;
}

int main() {
    auto startTime = std::chrono::high_resolution_clock::now();
    long long result = performHeavyTask();
    auto endTime = std::chrono::high_resolution_clock::now();

    auto elapsedTime = std::chrono::duration_cast<std::chrono::milliseconds>(endTime - startTime).count();

    std::cout << "Kết quả: " << result << std::endl;
    std::cout << "Thời gian thực thi: " << elapsedTime << "ms" << std::endl;

    return 0;
}
